chirp/drivers/ksun_m6.py

```diff
@@ -0,0 +1,477 @@
+
+    # Do an upload of the radio to the serial port
+    def sync_out(self):
+        do_upload(self)
+
+    # Convert the raw byte array into a memory object structure
+    def process_mmap(self):
+        self._memobj = bitwise.parse(MEM_FORMAT, self._mmap)
+
+    # Return a raw representation of the memory object, which
+    # is very helpful for development
+    def get_raw_memory(self, number):
+        return repr(self._memobj.memory[number-1])
+
+    # Extract a high-level memory object from the low-level memory map
+    # This is called to populate a memory in the UI
+    def get_memory(self, number):
+        _mem = self._memobj.memory[number-1]
+
+        mem = chirp_common.Memory()
+        mem.number = number
+
+        if _mem.freq.get_raw() == bytes([255] * 5):
+            mem.empty = True
+        else:
+            rx_freq = ((_mem.freq[0] << 12)
+                       + (_mem.freq[1] << 4) + (_mem.freq[2] >> 4))
+            tx_freq = (((_mem.freq[2] & 0xF) << 16)
+                       + (_mem.freq[3] << 8) + _mem.freq[4])
+
+            mem.freq = int(rx_freq / 2000 * 1000000)
+            mem.offset = int(tx_freq / 2000 * 1000000)
+
+            if _mem.no_tx:
+                mem.duplex = "off"
+            elif rx_freq != tx_freq:
+                mem.duplex = "split"
+
+            chirp_common.split_tone_decode(
+                mem,
+                mem_to_triplet(_mem.tx_tone, _mem.tx_code),
+                mem_to_triplet(_mem.rx_tone, _mem.rx_code))
+
+        mem.mode = "NFM" if not mem.empty and _mem.nfm else "FM"
+        mem.power = POWER_LIST[int(not mem.empty and _mem.low_pwr)]
+        mem.skip = "S" if not mem.empty and _mem.skip else ""
+
+        mem.extra = RadioSettingGroup("Extra", "extra")
+
+        hopping = False if mem.empty else _mem.hopping
+        rsv = RadioSettingValueBoolean(hopping)
+        rs = RadioSetting("hopping", "Hopping", rsv)
+        mem.extra.append(rs)
+
+        compander = False if mem.empty else _mem.compander
+        rsv = RadioSettingValueBoolean(compander)
+        rs = RadioSetting("compander", "Compander", rsv)
+        mem.extra.append(rs)
+
+        scrambler = False if mem.empty else _mem.scrambler
+        rsv = RadioSettingValueList(SCRAMBLER_LIST, SCRAMBLER_LIST[scrambler])
+        rs = RadioSetting("scrambler", _("Scrambler"), rsv)
+        mem.extra.append(rs)
+
+        return mem
+
+    # Store details about a high-level memory to the memory map
+    # This is called when a user edits a memory in the UI
+    def set_memory(self, mem):
+        _mem = self._memobj.memory[mem.number-1]
+
+        if mem.empty:
+            _mem.fill_raw(b"\xff")
+        else:
+            rx_freq = mem.freq
+            if mem.duplex == "split" and mem.offset:
+                tx_freq = mem.offset
+            else:
+                tx_freq = rx_freq
+
+            rx_freq = round(rx_freq / 1000000 * 2000)
+            tx_freq = round(tx_freq / 1000000 * 2000)
```

Comment:
This is intentional so that the target frequency is snapped to the closest representation and not to the lowest representable integer. This is something that bugs me with most other drivers.